src/mods.cpp

```diff
@@ -356,3 +353,78 @@ Json::Value getModstoreUrl(std::string url)
 }
 
 #endif
+
+ModMetadata::ModMetadata(const std::string &mod_name):
+	m_mod_name(mod_name),
+	m_modified(false)
+{
+	m_stringvars.clear();
+}
+
+void ModMetadata::clear()
+{
+	Metadata::clear();
+	m_modified = true;
+}
+
+bool ModMetadata::save(const std::string &root_path)
+{
+	Json::Value json;
+	for (StringMap::const_iterator it = m_stringvars.begin();
+			it != m_stringvars.end(); ++it) {
+		json[it->first] = it->second;
+	}
+
+	if (!fs::PathExists(root_path)) {
+		fs::CreateAllDirs(root_path);
+	} else {
+		if (!fs::IsDir(root_path)) {
+			errorstream << "ModMetadata[" << m_mod_name << "]: Unable to save. '"
+				<< root_path << "' is not a directory." << std::endl;
+			return false;
+		}
+	}
+
+	bool w_ok = fs::safeWriteToFile(root_path + DIR_DELIM + m_mod_name,
+		Json::FastWriter().write(json));
+
+	if (!w_ok) {
```

Comment:
would prefer the ! to be switched (ie, switch the blocks and invert the condition